packages/yoastseo/src/scoring/assessments/seo/KeywordDensityAssessment.js

```diff
@@ -331,12 +328,37 @@ class KeywordDensityAssessment extends Assessment {
 		if ( customApplicabilityConfig ) {
 			this._config.applicableIfTextLongerThan = customApplicabilityConfig;
 		}
-		let text = paper.getText();
-		text = removeHtmlBlocks( text );
-		const textLength = customCountLength ? customCountLength( text ) : getWords( text ).length;
+
+		const textLength = customCountLength ? customCountLength( paper.getText() ) : getAllWordsFromTree( paper ).length;
 
 		return paper.hasText() && paper.hasKeyword() && textLength >= this._config.applicableIfTextLongerThan;
 	}
 }
 
-export default KeywordDensityAssessment;
+/**
+ * This assessment checks if the keyphrase density is within the recommended range.
+ * KeywordDensityAssessment was the previous name for KeyphraseDensityAssessment (hence the name of this file).
+ * We keep (and expose) this assessment for backwards compatibility.
+ *
+ * @deprecated Since version 20.12 Use KeyphraseDensityAssessment instead.
```

Comment:
The version here could change depending on our plan to ship this change

Reply:
See comment above.